src/main/java/org/broadinstitute/hellbender/engine/ReadsDataSource.java

```diff
@@ -155,6 +161,16 @@ public ReadsDataSource( final List<Path> samPaths, final List<Path> samIndices,
         int samCount = 0;
         for ( final Path samPath : samPaths ) {
             // Ensure each file can be read
+            Function<SeekableByteChannel, SeekableByteChannel> wrapper = Function.identity();
+            Function<SeekableByteChannel, SeekableByteChannel> indexWrapper = Function.identity();
+            if (samPath.toUri().getScheme().equals("gs")) {
```

Comment:
Use `BucketUtils.isCloudStorageUrl()` instead (add a new overload of it that takes a `Path`, plus a unit test)